index.js

```diff
@@ -1,137 +1,145 @@
+      }
+    }
 
-  let shouldThrow = true
-  if ('throws' in options) {
-    shouldThrow = options.throws
-  }
+    const str = JSON.stringify(obj, options ? options.replacer : null, spaces)
 
-  try {
-    let content = fs.readFileSync(file, options)
-    content = stripBom(content)
-    return JSON.parse(content, options.reviver)
-  } catch (err) {
-    if (shouldThrow) {
-      err.message = `${file}: ${err.message}`
-      throw err
-    } else {
-      return null
-    }
+    return str.replace(/\n/g, EOL) + EOL
   }
-}
 
-function stringify (obj, options) {
-  let spaces
-  let EOL = '\n'
-  if (typeof options === 'object' && options !== null) {
-    if (options.spaces) {
-      spaces = options.spaces
+  function writeFileWithCallback (file, obj, options, callback) {
+    if (callback == null) {
+      callback = options
+      options = {}
     }
-    if (options.EOL) {
-      EOL = options.EOL
-    }
-  }
+    options = options || {}
+    const fs = options.fs || _fs
 
-  const str = JSON.stringify(obj, options ? options.replacer : null, spaces)
-
-  return str.replace(/\n/g, EOL) + EOL
-}
+    let str = ''
+    try {
+      str = stringify(obj, options)
+    } catch (err) {
+      return callback(err, null)
+    }
 
-function writeFileWithCallback (file, obj, options, callback) {
-  if (callback == null) {
-    callback = options
-    options = {}
+    fs.writeFile(file, str, options, callback)
   }
-  options = options || {}
-  const fs = options.fs || _fs
-
-  let str = ''
-  try {
-    str = stringify(obj, options)
-  } catch (err) {
-    return callback(err, null)
-  }
-
-  fs.writeFile(file, str, options, callback)
-}
 
-const writeFile = universalify.fromCallback(writeFileWithCallback)
+  const writeFile = universalify.fromCallback(writeFileWithCallback)
 
-function writeFileSync (file, obj, options) {
-  options = options || {}
-  const fs = options.fs || _fs
+  function writeFileSync (file, obj, options) {
+    options = options || {}
+    const fs = options.fs || _fs
 
-  const str = stringify(obj, options)
-  // not sure if fs.writeFileSync returns anything, but just in case
-  return fs.writeFileSync(file, str, options)
-}
+    const str = stringify(obj, options)
+    // not sure if fs.writeFileSync returns anything, but just in case
+    return fs.writeFileSync(file, str, options)
+  }
 
-function stripBom (content) {
+  function stripBom (content) {
   // we do this because JSON.parse would convert it to a utf8 string if encoding wasn't specified
-  if (Buffer.isBuffer(content)) content = content.toString('utf8')
-  content = content.replace(/^\uFEFF/, '')
-  return content
-}
+    if (Buffer.isBuffer(content)) content = content.toString('utf8')
+    content = content.replace(/^\uFEFF/, '')
+    return content
+  }
+
+  return {
+    readFile,
+    readFileSync,
+    writeFile,
+    writeFileSync
+  }
+})
 
-const jsonfile = {
-  readFile,
-  readFileSync,
-  writeFile,
-  writeFileSync
+function JsonFile (fs = realFs) {
```

Comment:
I'm curious why a function/factory pattern was used here instead of an actual `class`?

Reply:
A class would have nothing to do here, because it would have nothing on its prototype. This is equivalent, but if a class looks better for the codebase, I can refactor it.